### ISSUE TITLE
sqlite dependency issue

### ISSUE THREAD

**Issue by macheeto** [lmenezes/cerebro#121]
Hello,
Cerebro fails to start on Alpine Linux because sqlite has a problem:

Failed to load native library:sqlite-3.16.1-df8c3474-b137-4042-92ad-30e88cfbe642-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: /tmp/sqlite-3.16.1-df8c3474-b137-4042-92ad-30e88cfbe642-libsqlitejdbc.so: Error relocating /tmp/sqlite-3.16.1-df8c3474-b137-4042-92ad-30e88cfbe642-libsqlitejdbc.so: __isnan: symbol not found
Oops, cannot start the server.
java.sql.SQLTimeoutException: Timeout after 1000ms of waiting for a connection.
	at com.zaxxer.hikari.pool.BaseHikariPool.getConnection(BaseHikariPool.java:227)
	at com.zaxxer.hikari.pool.BaseHikariPool.getConnection(BaseHikariPool.java:182)
	at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:93)
	at slick.jdbc.hikaricp.HikariCPJdbcDataSource.createConnection(HikariCPJdbcDataSource.scala:12)
	at play.api.db.slick.evolutions.internal.DBApiAdapter$DatabaseAdapter.getConnection(DBApiAdapter.scala:57)
	at play.api.db.slick.evolutions.internal.DBApiAdapter$DatabaseAdapter.getConnection(DBApiAdapter.scala:60)
	at play.api.db.evolutions.DatabaseEvolutions.databaseEvolutions(EvolutionsApi.scala:124)
	at play.api.db.evolutions.DatabaseEvolutions.scripts(EvolutionsApi.scala:102)
	at play.api.db.evolutions.DatabaseEvolutions.scripts(EvolutionsApi.scala:117)
	at play.api.db.evolutions.DefaultEvolutionsApi.scripts(EvolutionsApi.scala:82)
	at play.api.db.evolutions.ApplicationEvolutions$$anonfun$play$api$db$evolutions$ApplicationEvolutions$$runEvolutions$1.apply$mcV$sp(ApplicationEvolutions.scala:53)

libsqlitejdbc.so: __isnan: symbol not found <<<

I'm using the official Docker image for Elasticsearch (docker.elastic.co/elasticsearch/elasticsearch:5.2.2) which is built on Alpine Linux and even after adding the sqlite dependency (apk add sqlite) cerebro fails to start.
How can I get it going?
Thanks!

**Comment by macheeto** [comment]
This error seems to be related to this issue with sqlite:
https://github.com/xerial/sqlite-jdbc/issues/97

Someone suggested setting org.sqlite.tmpdir property but that didn't work for me:

export JAVA_OPTS=-Dorg.sqlite.tmpdir=/usr/share/elasticsearch/temp
./cerebro

(Same error.)

The sqlite version bundled with cerebro is org.xerial.sqlite-jdbc-3.16.1.jar, which supposedly supports that config setting.

**Comment by lmenezes** [comment]
@macheeto I really don't know how to fix this right now. I will try to look into it in the next few days and let you know if I do find something. Sorry :man_shrugging:

**Comment by littlesnitch** [comment]
I had following exception
`Oops, cannot start the server.
java.sql.SQLTimeoutException: Timeout after 1001ms of waiting for a connection.
        at com.zaxxer.hikari.pool.BaseHikariPool.getConnection(BaseHikariPool.java:227)
        at com.zaxxer.hikari.pool.BaseHikariPool.getConnection(BaseHikariPool.java:182)
        at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:93)
        at slick.jdbc.hikaricp.HikariCPJdbcDataSource.createConnection(HikariCPJdbcDataSource.scala:12)
        at play.api.db.slick.evolutions.internal.DBApiAdapter$DatabaseAdapter.getConnection(DBApiAdapter.scala:57)
        at play.api.db.slick.evolutions.internal.DBApiAdapter$DatabaseAdapter.getConnection(DBApiAdapter.scala:60)
        at play.api.db.evolutions.DatabaseEvolutions.databaseEvolutions(EvolutionsApi.scala:124)
        at play.api.db.evolutions.DatabaseEvolutions.scripts(EvolutionsApi.scala:102)
        at play.api.db.evolutions.DatabaseEvolutions.scripts(EvolutionsApi.scala:117)
        at play.api.db.evolutions.DefaultEvolutionsApi.scripts(EvolutionsApi.scala:82)
        at play.api.db.evolutions.ApplicationEvolutions$$anonfun$play$api$db$evolutions$ApplicationEvolutions$$runEvolutions$1.apply$mcV$sp(ApplicationEvolutions.scala:53)
        at play.api.db.evolutions.ApplicationEvolutions.withLock(ApplicationEvolutions.scala:99)
        at play.api.db.evolutions.ApplicationEvolutions.play$api$db$evolutions$ApplicationEvolutions$$runEvolutions(ApplicationEvolutions.scala:49)
        at play.api.db.evolutions.ApplicationEvolutions$$anonfun$start$1.apply(ApplicationEvolutions.scala:42)
        at play.api.db.evolutions.ApplicationEvolutions$$anonfun$start$1.apply(ApplicationEvolutions.scala:42)
        at scala.collection.immutable.Stream.foreach(Stream.scala:594)
        at play.api.db.evolutions.ApplicationEvolutions.start(ApplicationEvolutions.scala:42)
        at play.api.db.evolutions.ApplicationEvolutions.<init>(ApplicationEvolutions.scala:150)
        at play.api.db.evolutions.ApplicationEvolutionsProvider.get$lzycompute(EvolutionsModule.scala:53)
        at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:53)
        at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:44)
        at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
        at com.google.inject.internal.BoundProviderFactory.provision(BoundProviderFactory.java:72)
        at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
        at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:62)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
        at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:205)
        at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:199)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
        at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:199)
        at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:180)
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:110)
        at com.google.inject.Guice.createInjector(Guice.java:96)
        at com.google.inject.Guice.createInjector(Guice.java:84)
        at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:181)
        at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:123)
        at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21)
        at play.core.server.ProdServerStart$.start(ProdServerStart.scala:47)
        at play.core.server.ProdServerStart$.main(ProdServerStart.scala:22)
        at play.core.server.ProdServerStart.main(ProdServerStart.scala)
`
Seems to be like the same.
I could resolve it with that parameter that was mentioned in one of the last posts:
bin\cerebro.bat -Dhttp.port=9000 -Dorg.sqlite.tmpdir=d:\tmp
This resulted under windows with a working Cerebro version 0.6.3. I still tried a version 0.5.1, and there I didn't have this error.

**Comment by macheeto** [comment]
I was able to get Cerebro working by not using the official Elasticsearch Docker image. There is some kind of permissions-related problem when using that image, related to the tmp directory.
I tested installing Elasticsearch and Cerebro on a standalone Linux box and that worked OK.
It looks like the solution is to build a custom Docker image with Elasticsearch and Cerebro and not use the official Elasticsearch Docker image (docker.elastic.co/elasticsearch/elasticsearch:5.2.2).
